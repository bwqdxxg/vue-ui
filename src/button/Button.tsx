import { defineComponent } from 'vue'
import { createNamespace } from '../utils';
import './index.less'

const [name, bem] = createNamespace('button');

export type ButtonType =
  | 'default'
  | 'primary'
  | 'success'
  | 'warning'
  | 'danger';

export type ButtonSize = 'large' | 'normal' | 'small' | 'mini';

export default defineComponent({
  name,

  emits: ['click'],

  setup(props, { emit, slots }) {
    const renderText = () => {
      let text = slots.default()
      return <span>{text}</span>;
    };

    return () => {
      return (
        <button className="bx-button">{renderText()}</button>
      )
    }
  }
})
