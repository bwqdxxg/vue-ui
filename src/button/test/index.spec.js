import { mount } from '@vue/test-utils';
import Button from '..';

test('render bx button', () => {
  const wrapper = mount(Button);
  expect(wrapper.html()).toMatchSnapshot();
});
