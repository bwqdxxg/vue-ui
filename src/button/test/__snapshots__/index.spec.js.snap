// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render bx button 1`] = `
<button class="bx-button">
</button>
`;
